tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImplV2.java

```diff
@@ -303,7 +308,7 @@ public static ReadHandle open(ScheduledExecutorService executor,
                                   VersionCheck versionCheck,
                                   long ledgerId, int readBufferSize, LedgerOffloaderStats offloaderStats,
                                   String managedLedgerName)
-            throws IOException {
+            throws IOException, BKException.BKNoSuchLedgerExistsOnMetadataServerException {
```

Comment:
Is it reasonable to throw `BKNoSuchLedgerExistsOnMetadataServerException ` ?

Reply:
In Bookkeeper, if ledger metadata isn't found on the metadata server, it throws BKNoSuchLedgerExistsOnMetadataServerException.

Likewise, when looking up the index (metadata) for the blob (ledger), this method should also throw the same exception for consistent handling in ManagerLedger.

Reply:
But from my understanding, 
```java
Blob blob = blobStore.getBlob(bucket, key, new GetOptions().range(startRange, endRange));
```
the `blob` should be the offloaded ledger, if `blob == null`, shouldn't it be means that the ledger not found?

And I've checked bookkeeper, see https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/LedgerDescriptor.java#L47
```java
    static LedgerDescriptor createReadOnly(long ledgerId,
                                           LedgerStorage ledgerStorage)
            throws IOException, Bookie.NoLedgerException {
        if (!ledgerStorage.ledgerExists(ledgerId)) {
            throw new Bookie.NoLedgerException(ledgerId);
        }
        return new LedgerDescriptorReadOnlyImpl(ledgerId, ledgerStorage);
    }
```
If the ledger id isn't in the `ledgerStorage`, it throws `NoLedgerException`

Reply:
And according to https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1955-L1962

It should be the ledger exists in the metadata server, but not found in the tiered-storage, am I right?

Reply:
Make sense to me, fixed.